data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java

```diff
@@ -27,12 +36,40 @@ public BeaconBlockHandler(ChainStorageClient client) {
 
   @Override
   public String getPath() {
-    return "/beacon/block";
+    return "/beacon/block/";
   }
 
   @Override
   public Object handleRequest(RequestParams param) {
-    Bytes32 root = Bytes32.fromHexString(param.getQueryParam("root"));
-    return client.getStore() != null ? client.getStore().getBlock(root) : null;
+    Map<String, List<String>> queryParamMap = param.getQueryParamMap();
+    Map<String, Object> jsonObject = new HashMap<>();
+    BeaconBlock block;
+    Bytes32 blockRoot;
+    if (queryParamMap.containsKey("root")) {
+      Bytes32 root = Bytes32.fromHexString(param.getQueryParam("root"));
+      return client.getStore() != null ? client.getStore().getBlock(root) : null;
+    } else if (queryParamMap.containsKey("epoch")) {
+      UnsignedLong epoch = UnsignedLong.valueOf(param.getQueryParam("epoch"));
+      Optional<Bytes32> blockRootAtSlot =
+          client.getBlockRootBySlot(compute_start_slot_at_epoch(epoch));
+      blockRoot = blockRootAtSlot.orElse(null);
+      block =
+          client.getStore() != null && blockRootAtSlot.isPresent()
+              ? client.getStore().getBlock(blockRootAtSlot.get())
+              : null;
+    } else if (queryParamMap.containsKey("slot")) {
+      UnsignedLong slot = UnsignedLong.valueOf(param.getQueryParam("slot"));
+      Optional<Bytes32> blockRootAtSlot = client.getBlockRootBySlot(slot);
+      blockRoot = blockRootAtSlot.orElse(null);
+      block =
+          client.getStore() != null && blockRootAtSlot.isPresent()
+              ? client.getStore().getBlock(blockRootAtSlot.get())
+              : null;
+    } else {
+      throw new ForbiddenResponse("Off limits.");
```

Comment:
I think we should avoid setting the precedent that Javalin APIs leak into here - it should be transport agnostic.  Maybe just return `null` for now and we can sort out how to handle more meaningful error response codes when we properly revisit the REST APIs.

Reply:
Makes sense to me.